ophyd/device.py

```diff
@@ -264,8 +325,87 @@ def sub_value(self, func):
         'Value subscription decorator'
         return self.subscriptions('value')(func)
 
+    @classmethod
+    def create(
+        cpt_cls: Type[Component],
+        cls: Type[K],
+        suffix: Optional[str] = None,
+        *,
+        lazy: Optional[bool] = None,
+        trigger_value: Optional[Any] = None,
+        add_prefix: Optional[Sequence[str]] = None,
+        doc: Optional[str] = None,
+        kind: Union[str, Kind] = Kind.normal,
+        **kwargs
+    ) -> Component[K]:
+        """
+        A wrapper for ``Component`` for easy migration to Component[T] generic
+        support.
+
+        With ``Component[T]``, static code analyzers such as mypy or pyright
+        will be much happier with ophyd Device code in general.
+
+        Parameters
+        ----------
+        kls : class
+            Class of signal to create.  The required signature of
+            `cls.__init__` is
+            If `suffix` is given::
+
+                def __init__(self, pv_name, parent=None, **kwargs):
+
+            Or, if suffix is None::
 
-class FormattedComponent(Component):
+                def __init__(self, parent=None, **kwargs):
+
+            The class may have a `wait_for_connection()` which is called during
+            the component instance creation.
+
+        suffix : str, optional
+            The PV suffix, which gets appended onto ``parent.prefix`` to
+            generate the final PV that the instance component will bind to.
+            Also see ``add_prefix``
+
+        lazy : bool, optional
+            Lazily instantiate the signal. If ``False``, the signal will be
+            instantiated upon component instantiation. Defaults to
+            ``component.lazy_default``.
+
+        trigger_value : any, optional
+            Mark as a signal to be set on trigger. The value is sent to the
+            signal at trigger time.
+
+        add_prefix : sequence, optional
+            Keys in the kwargs to prefix with the Device PV prefix during
+            creation of the component instance.
+            Defaults to ``('suffix', 'write_pv', )``
+
+        doc : str, optional
+            string to attach to component DvcClass.component.__doc__
+
+        **kwargs :
+            Unrecognized keyword arguments will be passed directly to the
+            component class initializer.
+
+        Returns
+        -------
+        Component[cls]
+            A component instance which specifies the creation of ``cls`` when
+            the parent device is made.
+        """
+        return cpt_cls[cls](
+            cls,
+            suffix,
+            lazy=lazy,
+            trigger_value=trigger_value,
+            add_prefix=add_prefix,
+            doc=doc,
+            kind=kind,
+            **kwargs
+        )
+
+
+class FormattedComponent(Component[K]):
```

Comment:
Why does `ADComponent` have duplicated annotations from `Component`, while `FormattedComponent does not?

Reply:
Solely for sphinx (note the failed docs build prior to [aaa7183](https://github.com/bluesky/ophyd/pull/1053/commits/aaa718335f5f8ac67d00d806527869790d654d9c)); I believe it to be a bug and this was the quickest way forward.